src/main/java/core/basesyntax/figuresuppliers/FigureSupplier.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.figuresuppliers;
+
+import core.basesyntax.color.Color;
+import core.basesyntax.color.ColorSupplier;
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    static final Color DEFAULT_COLOR = Color.WHITE;
+    static final int MAX_EDGE = 10;
+    static final int ANNEX = 1;
+    static final int DEFAULT_RADIUS = 10;
+
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int sizeOne = random.nextInt(MAX_EDGE) + ANNEX;
+        int index = new Random().nextInt(FigureType.values().length);
+        FigureType figureType = FigureType.values()[index];
+        Color randomColor = colorSupplier.getRandomColor();
+
+        switch (figureType) {
+            case SQUARE:
+                return new Square(randomColor, sizeOne);
+            case RECTANGLE:
+                int sizeTwo = random.nextInt(MAX_EDGE) + ANNEX;
+                return new Rectangle(randomColor, sizeOne, sizeTwo);
+            case RIGHT_TRIANGLE:
+                sizeTwo = random.nextInt(MAX_EDGE) + ANNEX;
+                return new RightTriangle(randomColor, sizeOne, sizeTwo);
+            case CIRCLE:
+                return new Circle(randomColor, sizeOne);
+            case ISOSCELES_TRAPEZOID:
+                sizeTwo = random.nextInt(MAX_EDGE) + ANNEX;
+                int sizeThree = random.nextInt(MAX_EDGE) + ANNEX;
+                return new IsoscelesTrapezoid(randomColor,
+                        sizeOne, sizeTwo, sizeThree);
+            default: return getDefaultFigure();
+        }

```

Comment:
This part has a problem with code duplication. You generate random sizes using one formula. Let's imagine tomorrow you'll need to change this formula, then you'll have to go over each line and make a change. It's insufficient. In this case, you can create a separate method for this computation and reuse it.